### PR TITLE
Cleanup/remove unecessary

### LINKED ISSUE #3713: Images should use less memory
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

The image handling code on `develop` uses a lot of memory. Current count is:

* cache contents of the file (to close the reader)
* The tee reader populates a buffer which may duplicate the above to support reading image metadata
* The image data is then decided into pixels which are also cached
* The image is then rendered to a texture cache duplicating the pixel info

we should be able to remove 3/5 of those caches. At least 2 should be cleared without need for any api changes.

### How to reproduce

Use `NewImageFromFile` (or resource, or reader etc)

### Screenshots

_No response_

### Example code

`NewImageFromReader(“myfile.png”)`

### Fyne version

develop

### Go compiler version

All

### Operating system and version

Any

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Bluebugs**
### Description:
Quick tidying around image.

For #3713 

### Checklist:
- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Comment by andydotxyz** [comment]
It looks like a useful change, but it does not fix the memory issues that it marks as done.
More duplication exists in the `canvas.Image` load/decode code.

**Comment by Bluebugs** [comment]
It is indeed a partial fix. Is there a different keyword in that case?

**Comment by andydotxyz** [comment]
> It is indeed a partial fix. Is there a different keyword in that case?

You just need to reference the #number and it will link. "fixes" specifically activates the issue close workflow.

**Comment by Bluebugs** [comment]
Thanks, it is corrected now.

**Comment by Jacalz** [comment]
I updated it to say "For #nr". I think that is how it is usually done :)

File: internal/painter/gl/texture.go [hunk 1]
```diff
@@ -113,15 +113,11 @@ func (p *painter) newGlCircleTexture(obj fyne.CanvasObject) Texture {
 func (p *painter) newGlImageTexture(obj fyne.CanvasObject) Texture {
 	img := obj.(*canvas.Image)
 
-	width := p.textureScale(img.Size().Width)
-	height := p.textureScale(img.Size().Height)
-
-	tex := paint.PaintImage(img, p.canvas, int(width), int(height))
```
Comment (andydotxyz) [inline internal/painter/gl/texture.go:119]: If this really isn't needed here then can you apply the same fix to the `software/draw.go` as well?
Then I don't think this `PaintImage` method is needed any more either...

**Review by andydotxyz — CHANGES_REQUESTED** [review]

File: internal/painter/gl/texture.go (hunk 1 above)
Comment (Bluebugs) [reply internal/painter/gl/texture.go:119]: The software backend does actually use PaintImage to do the scaling. You can see that the size being passed is actually different from the size of the image in the software case and not in the case of GL.

Comment (andydotxyz) [reply internal/painter/gl/texture.go:119]: Isn't it the same here? PaintImage is being given the current size of the object, not the size of the image pixels

Comment (Bluebugs) [reply internal/painter/gl/texture.go:119]: It isn't exactly the same, but I think you are right. I will drop that part of the change for now and come back to it later.

**Comment by andydotxyz** [comment]
I'm looking at the change and it looks like the `Refresh` method sets the object field through `updateReader` then sets it to `nil` inside the same method.
I could be wrong but would it not be tidier to remove that as an object field and just handle it locally?

**Comment by Bluebugs** [comment]
Seems doable, will look into it.

**Comment by Bluebugs** [comment]
@andydotxyz that worked out.

**Review by andydotxyz — APPROVED** [review]
Thanks

**Pull request merged**